sympy/matrices/matrices.py

```diff
@@ -3888,59 +3891,164 @@ def LUdecomposition(self, iszerofunc=_iszero):
+                    iszeropivot = iszerofunc(lu[candidate_pivot_row,
+                                                candidate_pivot_col])
+                    if iszeropivot is None:
+                        iszeropivot = False
+                        break
+                    if not iszeropivot:
+                        break
+                if not iszeropivot:
+                    break
+            pivot_col = candidate_pivot_col
+
+            if rankcheck and pivot_col != pivot_row:
+                # All entries including and below the pivot position are
+                # zero, which indicates that the rank of the matrix is
+                # strictly less than min(num rows, num cols)
+                # Mimic behavior of previous implementation, by throwing a
+                # ValueError.
+                raise ValueError("Rank of matrix is strictly less than"
+                                 " number of rows or columns."
+                                 " Pass keyword argument"
+                                 " rankcheck=False to compute"
+                                 " the LU decomposition of this matrix.")
+
+            elif iszeropivot:
+                # iszeropivot is True after pivot search has completed
+                # indicates that there are no more entries to zero,
+                # so Gaussian elimination is complete.
+                return lu, row_swaps
+
+            if pivot_row != candidate_pivot_row:
+                # Row swap book keeping:
+                # Record which rows were swapped.
+                # Update stored portion of L factor by multiplying L on the
+                # left and right with the current permutation.
+                # Swap rows of U.
+                row_swaps.append([pivot_row, candidate_pivot_row])
+
+                # Update L.
+                for col in range(0, pivot_row):
+                    lu[pivot_row, col], lu[candidate_pivot_row, col] =\
+                        lu[candidate_pivot_row, col], lu[pivot_row, col]
+
+                # Swap rows of U in the pivot column.
+                lu[pivot_row, pivot_col] = lu[candidate_pivot_row, pivot_col]
+                lu[candidate_pivot_row, pivot_col] = S.Zero
+
+                for col in range(pivot_col+1, lu.cols):
+                    tmp = lu[pivot_row, col]
+                    lu[pivot_row, col] = lu[candidate_pivot_row, col]
+                    lu[candidate_pivot_row, col] = tmp
+
+            for row in range(pivot_row + 1, lu.rows):
+                # Store factors of L in the subcolumn below
+                # (pivot_row, pivot_row).
+                lu[row, pivot_row] =\
+                    lu[row, pivot_col]/lu[pivot_row, pivot_col]
+
+                # Add multiple of pivot row to row below it.
+                # Two loops to handle case where pivot_col > pivot_row
+
+                for col in range(pivot_row + 1 if pivot_row == pivot_col
+                                 else pivot_col, lu.cols):
```

Comment:
This is hard to parse with a big `if...else` clause in the argument list.  Can you save this to a well-named variable and then use it?